src/components/CopyStateLinkButton/index.tsx

```diff
@@ -0,0 +1,139 @@
+                    setSelectedCountryInSidebar({ name: newViewValues.name }),
+                );
+                dispatch(setPopup({ isOpen: true, countryName: country.name }));
+
+                return;
+            }
+        }
+
+        dispatch(setPopup({ isOpen: false, countryName: '' }));
+        dispatch(setSelectedCountryInSidebar(null));
+        setSnackbarAlertOpen(newViewValues.name !== 'worldwide');
+    }, [location.search, countriesData]);
+
+    const [copyHandler, setCopyHandler] = useState({
+        message: `Copy link to ${onWhichContainer}`,
+        isCopying: false,
+    });
+
+    const [snackbarAlertOpen, setSnackbarAlertOpen] = useState(false);
+
+    useEffect(() => {
+        if (!snackbarAlertOpen) return;
+        setTimeout(() => {
+            setSnackbarAlertOpen(false);
+        }, 3000);
+    }, [snackbarAlertOpen]);
+
+    const handleCopyLinkButton = () => {
+        if (copyHandler.isCopying) return;
+
+        const center = map?.current?.getCenter().toArray();
+        const zoom = map?.current?.getZoom();
+
+        const countryName = selectedCountry
+            ? selectedCountry.name
+            : 'worldwide';
+
+        if (onWhichContainer === 'chart') {
+            navigator.clipboard.writeText(
+                `${window.location.href}?name=${countryName}&startDate=${chartDatePeriod[0]}&endDate=${chartDatePeriod[1]}`,
+            );
+        } else {
+            navigator.clipboard.writeText(
+                `${
+                    window.location.href
+                }?name=${countryName}&currDate=${timeseriesDates.indexOf(
+                    currentDate || timeseriesDates[timeseriesDates.length - 1],
+                )}${
+                    center &&
+                    zoom &&
+                    '&lng=' + center[0] + '&lat=' + center[1] + '&zoom=' + zoom
+                }`,
+            );
+        }
+        setCopyHandler({ message: 'Copied!', isCopying: true });
+
+        setTimeout(() => {
+            setCopyHandler({
+                message: `Copy link to ${onWhichContainer}`,
+                isCopying: false,
+            });
+        }, 2000);
+    };
+
+    return (
+        <>
+            <CopyStateLinkButtonContainer
+                color="primary"
+                variant="extended"
+                onClick={handleCopyLinkButton}
+            >
+                {' '}
+                {copyHandler.isCopying ? <DoneIcon /> : <LinkIcon />}
+                {copyHandler.message}
+            </CopyStateLinkButtonContainer>
+            <Snackbar
+                open={snackbarAlertOpen}
+                // open={true}
+                onClose={() => setSnackbarAlertOpen(false)}
+                anchorOrigin={{ horizontal: 'center', vertical: 'top' }}
+                sx={{ height: '100%' }}
+            >
+                <Alert severity="error" variant="filled">
+                    Unfortunately, there is no data from the country that u
```

Comment:
I think it would be better to change `that u selected` -> `that you have selected`